src/mul/mul.jl

```diff
@@ -17,6 +17,30 @@ function subspace_mul!(st::AbstractVector{T}, comspace, U, subspace, offset=0) w
     end
 end
 
+# specialize on the X gate
+function subspace_mul!(st::AbstractVector{T}, loc::Int, U::Val{:X_test}, offset=0) where T
+    n = log2dim(st)
+    loc_bit = 1 << (loc-1)
+    k = 0
+
+    while true
```

Comment:
please NEVER use `while true` in the code

Reply:
maybe you want to consider read the previous generated code from codegen for X gate first and understand that then modifiy it to multithreaded version instead of reinventing the wheels.

Reply:
Indeed, the `while` loop is not amenable to SIMD. I've updated `subspace_mul!` strictly according to `codegen`, with loop unrolling. I've also updated the benchmark results.